### PR TITLE
improved documentation according to #480

### PR DESCRIPTION
**Pull request by loicjaouen**
follow-up on the difficulties to run salsah's test as discussed in #480

### CODE REVIEW

File: docs/rst/salsah/development/build-process.rst [hunk 1]
```diff
@@ -78,18 +78,24 @@ To shut down the SALSAH server:
...
-Make sure you've started Fuseki and the API server as shown above.
 In order to run the tests, the Selenium driver for Chrome has to be installed.
-Please download it from `here <https://sites.google.com/a/chromium.org/chromedriver/downloads>`_ and save it
-as ``salsah/lib/chromedriver``. Also, please make sure to start the API server with the
-``allowResetTriplestoreContentOperationOverHTTP`` flag. For more information about this flag, see :ref:`webapi-server-startup-flags`
 
-Then at the SBT prompt:
+It is architecture-dependant, please go to ``salsah/lib/chromedriver`` and unzip there the one that matches your architecture, or download it from `here <https://sites.google.com/a/chromium.org/chromedriver/downloads>`_ and install it in this directory.
+Then, make sure that:
+    - Fuseki was started as shown above,
```
Comment (tobiasschweizer) [inline docs/rst/salsah/development/build-process.rst]: When buidling the docs I get this warning: `docs/rst/salsah/development/build-process.rst:85: ERROR: Unexpected indentation`

I think you have to put a newline before you begin the list: <http://docutils.sourceforge.net/docs/user/rst/quickref.html#bullet-lists>

File: docs/rst/salsah/development/build-process.rst [hunk 2]
```diff
@@ -78,18 +78,24 @@ To shut down the SALSAH server:
...
 
 ::
 
+    $ cd KNORA_PROJECT_DIRECTORY/salsah
+    $ sbt
+    > compile
+    > re-start
     > test
 
+Note: in case nothing happens, make sure you wait long enough; Salsah reports some errors only after a timeout of 1 minute.
```
Comment (tobiasschweizer) [inline docs/rst/salsah/development/build-process.rst]: What do you mean by that? If nothing happens, something is clearly wrong!

Either salsah is not running or webapi is not running or was not started with the correct flag which allows for reloading the test data over HTTP.

File: docs/rst/salsah/development/build-process.rst (hunk 1 above)
Comment (tobiasschweizer) [inline docs/rst/salsah/development/build-process.rst]: or graphdb ...

Could you mention that Sipi should be running with `sipi.knora-test-config.lua`?

File: docs/rst/salsah/development/build-process.rst (hunk 2 above)
Comment (loicjaouen) [inline docs/rst/salsah/development/build-process.rst]: thanks for the review.
I am re-phrasing the text.

Comment (tobiasschweizer) [inline docs/rst/salsah/development/build-process.rst]: Sorry, but I still do not understand why the tests won't pass for you.

Comment (loicjaouen) [inline docs/rst/salsah/development/build-process.rst]: Why the tests failed specifically for me: I guess I did something wrong, I can start again to check.

But this line here is not specifically for me; as we do changes, we expect to break things and see tests failing. I just want to warn that in the case of salsah the failure notification can take up to one minute to come.

Comment (tobiasschweizer) [inline docs/rst/salsah/development/build-process.rst]: Maybe salsah fails when attempting to reload the test data?

**Comment by benjamingeer** [comment]
> Maybe salsah fails when attempting to reload the test data?

Maybe Loïc is talking about the timeout error that Salsah prints if it doesn't get any input for 60 seconds?

**Comment by loicjaouen** [comment]
no ;)

all right, let me try to reproduce that

**Comment by tobiasschweizer** [comment]
> Maybe Loïc is talking about the timeout error that Salsah prints if it doesn't get any input for 60 seconds?

we should get rid of that one, it annoys me

**Comment by loicjaouen** [comment]
If you look at [imagebase.js](https://github.com/dhlab-basel/Knora/blob/develop/salsah/src/public/js/imagebase.js#L2487) there is a timeout of 50s.

I don't know if it is linked but if you try to start the tests without starting salsah, you get to wait 1 minute:

```
[info] Loading project definition from /Users/ljaouen/Soft/Code/github/KnoraBasle/salsah/project
[info] Set current project to salsah (in build file:/Users/ljaouen/Soft/Code/github/KnoraBasle/salsah/)
[info] SearchAndEditSpec:
[info] The SALSAH home page
[info] - should load test data (52 seconds, 163 milliseconds)
[info] - should have the correct title *** FAILED *** (277 milliseconds)
[info]   "[localhost]" was not equal to "[System for Annotation and Linkage of Sources in Arts and Humanities]" (SearchAndEditSpec.scala:97)
```

Here is another 1mn TO:

```
error
Internal Server Error
{"status":4,"error":"akka.pattern.AskTimeoutException: Ask timed out on [ActorSelection[Anchor(akka://webapi/), Path(/user/responderManager)]] after [60000 ms]. Sender[null] sent message of type \"org.knora.webapi.messages.v1.responder.resourcemessages.ResourceCreateRequestV1\"."}
```

But this is not yet the one I was thinking of is:

**Comment by benjamingeer** [comment]
> I don't know if it is linked but if you try to start the tests without starting salsah, you get to wait 1 minute

This is just because the first test takes one minute, as it says:

> [info] - should load test data (52 seconds, 163 milliseconds)

That test passes (the test data is loaded). Then the second test fails because Salsah isn't running, so the HTML page is not loaded:

> [info] - should have the correct title *** FAILED *** (277 milliseconds)

**Comment by loicjaouen** [comment]
apparently, I didn't finish my last sentence, what I was talking about was this, you create a resource with a file, here an anything picture, where the user can upload to sipi and create the resource in knora, but when viewing the created resource, knora doesn't grant sipi the right to display the file to the user, then you get an immediate unauthorized response from sipi, but salsah still waits 1 minute to display the pop-up `There is a problem loading the image "http://localhost:1024/knora/GahaXSKufcE-G12XchXmiMp.jpx/full/752,500/0/default.jpg`.

capture d ecran 2017-04-18 a 10 23 30

capture d ecran 2017-04-18 a 10 23 44

**Comment by loicjaouen** [comment]
i will move that to issues elsewhere, but concerning this PR, do the changes make sense now and can we merge and sweep up?

**Comment by tobiasschweizer** [comment]
Now I get what you meant!

Yes, please open an issue for that.

**Comment by loicjaouen** [comment]
here is the issue #487 
i'll merge as soon as the checks pass.
Thanks :)

**Pull request merged**